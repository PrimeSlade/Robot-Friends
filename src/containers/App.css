.text{
    font-weight: 600;
    color: #0ccac4;
}