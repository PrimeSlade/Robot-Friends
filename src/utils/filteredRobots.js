export const filteredRobots = (robots, searchField) => {
  return robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );
};
