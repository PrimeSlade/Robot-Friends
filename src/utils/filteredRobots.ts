import { Robot } from "../type/Types";

export const filteredRobots = (
  robots: Array<Robot>,
  searchField: string
): Array<Robot> => {
  return robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );
};
