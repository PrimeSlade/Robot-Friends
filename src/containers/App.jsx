import { robots } from "../robots";
import "./App.css";
import SearchBox from "../components/SearchBox";
import { lazy, Suspense, useState } from "react";
import Scroll from "../components/Scroll";

const CardList = lazy(() => import("../components/CardList"));

function App() {
  const [searchField, setSearchField] = useState(""); //defualt value is ""

  const filterdRobots = robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );

  const [counter, setCounter] = useState(0);

  return (
    <>
      <div className="tc">
        <h1 className="text">RobotFriends</h1>
        <button onClick={() => setCounter((c) => c + 1)} className="mb3">
          Click {counter}
        </button>
        <SearchBox setSearchField={setSearchField} />
        <Scroll>
          <Suspense
            fallback={<div className="text mt3">Loading robots...</div>}
          >
            <CardList robots={filterdRobots} />
          </Suspense>
        </Scroll>
      </div>
    </>
  );
}

export default App;
