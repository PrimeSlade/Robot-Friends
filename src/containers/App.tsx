import { robots } from "../robots";
import "./App.css";
import SearchBox from "../components/SearchBox";
import { lazy, Suspense, useState } from "react";
import Scroll from "../components/Scroll";
import CounterButton from "../components/CounterButton";
import { filteredRobots } from "../utils/filteredRobots";
import React from "react";

const CardList = lazy(() => import("../components/CardList"));

function App() {
  const [searchField, setSearchField] = useState(""); //defualt value is ""

  const filterdRobots = filteredRobots(robots, searchField);

  return (
    <>
      <div className="tc">
        <h1 className="text">RobotFriends</h1>
        {/* <CounterButton /> */}
        <SearchBox setSearchField={setSearchField} />
        <Scroll>
          <Suspense
            fallback={<div className="text mt3">Loading robots...</div>}
          >
            <CardList robots={filterdRobots} />
          </Suspense>
        </Scroll>
      </div>
    </>
  );
}

export default App;
