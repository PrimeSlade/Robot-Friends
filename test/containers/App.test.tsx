import { render, waitFor, screen } from "@testing-library/react";

import App from "../../src/containers/App";
import { filteredRobots } from "../../src/utils/filteredRobots";
import { robots } from "../../src/robots";
import React from "react";
import { mockRobots } from "../__mocks__/mockRobots";

describe("App Component and utils tests", () => {
  test("testing app by using snapshot", async () => {
    const { asFragment } = render(<App />);

    // Wait for loading to complete
    await waitFor(() => {
      expect(screen.queryByText("Loading robots...")).not.toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  test("testing filtered func", () => {
    expect(filteredRobots(robots, "")).toEqual(robots);
    expect(filteredRobots(mockRobots, "one")).toEqual([mockRobots[0]]);
  });
});
