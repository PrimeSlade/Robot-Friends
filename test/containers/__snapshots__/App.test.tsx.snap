// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component and utils tests testing app by using snapshot 1`] = `
<DocumentFragment>
  <div
    class="tc"
  >
    <h1
      class="text"
    >
      RobotFriends
    </h1>
    <button
      class="mb3"
    >
      Click 0
    </button>
    <div>
      <input
        class="pa3 ba b--green bg-lightest-blue"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow: scroll; border: 1px solid black; height: 500px; margin-top: 50px;"
    >
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/1?200*200"
        />
        <div>
          <h2>
            Leanne Graham
          </h2>
          <p>
            Sincere@april.biz
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/2?200*200"
        />
        <div>
          <h2>
            Ervin Howell
          </h2>
          <p>
            Shanna@melissa.tv
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/3?200*200"
        />
        <div>
          <h2>
            Clementine Bauch
          </h2>
          <p>
            Nathan@yesenia.net
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/4?200*200"
        />
        <div>
          <h2>
            Patricia Lebsack
          </h2>
          <p>
            Julianne.OConner@kory.org
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/5?200*200"
        />
        <div>
          <h2>
            Chelsey Dietrich
          </h2>
          <p>
            Lucio_Hettinger@annie.ca
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/6?200*200"
        />
        <div>
          <h2>
            Mrs. Dennis Schulist
          </h2>
          <p>
            Karley_Dach@jasper.info
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/7?200*200"
        />
        <div>
          <h2>
            Kurtis Weissnat
          </h2>
          <p>
            Telly.Hoeger@billy.biz
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/8?200*200"
        />
        <div>
          <h2>
            Nicholas Runolfsdottir V
          </h2>
          <p>
            Sherwood@rosamond.me
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/9?200*200"
        />
        <div>
          <h2>
            Glenna Reichert
          </h2>
          <p>
            Chaim_McDermott@dana.io
          </p>
        </div>
      </div>
      <div
        class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
        data-testid="card"
      >
        <img
          alt="robot"
          src="https://robohash.org/10?200*200"
        />
        <div>
          <h2>
            Clementina DuBuque
          </h2>
          <p>
            Rey.Padberg@karina.biz
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
